src/interpreter_funcs.rs

```diff
@@ -199,7 +199,58 @@ impl Func for FuncImplSeparateIsolatedMeshes {
         let values = mesh_tools::separate_isolated_meshes(geometry);
 
         // FIXME: This returns a slice of Geometries. Return all of them
-        Value::Geometry(Arc::new(values[0].clone()))
+        let first_value = values
+            .into_iter()
+            .next()
+            .expect("Need at least one geometry");
+
+        Value::Geometry(Arc::new(first_value))
+    }
+}
+
+pub struct FuncImplLoopSubdivision;
+impl Func for FuncImplLoopSubdivision {
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[
+            ParamInfo {
+                ty: Ty::Geometry,
+                optional: false,
+            },
+            ParamInfo {
+                ty: Ty::Uint,
+                optional: false,
+            },
+        ]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Geometry
+    }
+
+    fn call(&self, args: &[Value]) -> Value {
+        let geometry = args[0].unwrap_refcounted_geometry();
+        let iterations = args[1].unwrap_uint();
+        if iterations == 0 {
+            return Value::Geometry(geometry);
+        }
+
+        let mut v2v_topology = mesh_topology_analysis::vertex_to_vertex_topology(&geometry);
+        let mut f2f_topology = mesh_topology_analysis::face_to_face_topology(&geometry);
+        let mut current_geometry =
+            mesh_smoothing::loop_subdivision(&geometry, &v2v_topology, &f2f_topology);
+
+        for _ in 1..iterations {
```

Comment:
Please, make sure no more than 3 iterations are requested in one operation because anything more can and will result in an enormous mesh and slow calculation times. 3 iterations are an industry standard and even that usually takes a long time (seconds) to calculate. If it's is not enough, it's always possible to stack another smoothing operation into the pipeline.

Reply:
Okay. I'll add a clamp for now, and we can provide a range for the UI later.